server/fleet/datastore.go

```diff
@@ -2030,6 +2030,8 @@ type Datastore interface {
 
 	CreateOrUpdateSoftwareTitleIcon(ctx context.Context, payload *UploadSoftwareTitleIconPayload) (*SoftwareTitleIcon, error)
 	GetSoftwareTitleIcon(ctx context.Context, teamID uint, titleID uint) (*SoftwareTitleIcon, error)
+	GetSoftwareIconsByTeamAndTitleIds(ctx context.Context, teamID uint, titleIDs []uint) ([]SoftwareTitleIcon, error)
```

Comment:
If we want to keep this the way it is (which is consistent with automatic policy lookups but not VPP app lookups), probably want to add a comment explaining why. Though given that you're having to build this into a map anyway in ListHostSoftware, thinking we want a map here instead.